src/fsharp/NameResolution.fs

```diff
@@ -701,6 +721,7 @@ let MakeNestedModuleRefs (modref: ModuleOrNamespaceRef) =
 //
 // Recursive because of "AutoOpen", i.e. adding a module reference may automatically open further modules
 let rec AddModuleOrNamespaceRefsToNameEnv g amap m root ad nenv (modrefs: ModuleOrNamespaceRef list) =
```

Comment:
Is this equivalent?

I guess yes if `modrefs` is empty we don't change `nenv`, but I'm not clear what `NameMap` module is and it seems `layerAdditive` was called before even if modrefs was empty.